src/common/Throttle.cc

```diff
@@ -651,43 +651,88 @@ void OrderedThrottle::complete_pending_ops(UNIQUE_LOCK_T(m_lock)& l) {
 }
 
 uint64_t TokenBucketThrottle::Bucket::get(uint64_t c) {
-  if (0 == max) {
+  if (0 == cir) {
     return 0;
   }
 
-  uint64_t got = 0;
-  if (remain >= c) {
-    // There is enough token in bucket, take c.
-    got = c;
-    remain -= c;
-  } else {
-    // There is not enough, take all remain.
+  uint64_t got;
+  if (remain < c) {
     got = remain;
     remain = 0;
+  } else {
+    got = c;
+    remain -= c;
```

Comment:
Any reason for this change?

Reply:
I rewrite the class Bucket. Not deliberately want to change it.